src/logic/contracts/safeContracts.ts

```diff
@@ -20,24 +20,33 @@ export const DEFAULT_FALLBACK_HANDLER_ADDRESS = '0xd5D82B6aDDc9027B22dCA772Aa68D
 export const SAFE_MASTER_COPY_ADDRESS_V10 = '0xb6029EA3B2c51D09a50B53CA8012FeEB05bDa35A'
 
 
-let proxyFactoryMaster
-let safeMaster
-
-const createGnosisSafeContract = (web3: Web3) => {
-  const gnosisSafe = contract(GnosisSafeSol)
-  gnosisSafe.setProvider(web3.currentProvider)
+let proxyFactoryMaster: GnosisSafeProxyFactory
+let safeMaster: GnosisSafe
 
-  return gnosisSafe
+/**
+ * Creates a Contract instance of the GnosisSafe contract
+ * @param {Web3} web3
+ * @param {ETHEREUM_NETWORK} networkId
+ */
+const createGnosisSafeContract = (web3: Web3, networkId: ETHEREUM_NETWORK) => {
+  const networks = GnosisSafeSol.networks
+  const contractAddress = networks[networkId]?.address ?? networks[ETHEREUM_NETWORK.MAINNET].address
```

Comment:
can we add a comment here to explain why we are using MAINNET address as a fallback?

Reply:
perhaps, in the future, we should think about adding the fallback address in the config file.

Reply:
Yes, I'll add the comment.